### DISCUSSION TITLE
ScriptKit is cool! - Here's how it is helping me with manual repetitive tasks.

### DISCUSSION THREAD

**Discussion by rvisharma** [johnlindquist/kit#108] [Share]
I had a use-case where the local development server needs the oauth token from a deployed application to be stored on a local file. Problem is that the expiry of token is short which interrupts my flow of work and then i have to again do the following steps..
- Open the deployed application
- Login in with credentials
- Snip the token from devtools
- Switch over local file and replace the token.

Although, it just takes one minute, but this task is dreadful to me.

My initial idea was to create a chrome extension which will snip the token from the website and will ping a local server endpoint which in turn will update the local-file with new token using `fs`

***Enter Scriptkit***

It does everything i needed with a solid framework to invoke 💪🏻

I'm using, Playwright for browser automation _(open -> login -> read token)_ and then using `fs` update the local file.
I can invoke it via terminal OR via prompt. It just works.

CLI | Prompt
------------ | -------------
![in-cli](https://user-images.githubusercontent.com/5556114/113440541-41d51700-940a-11eb-9562-3a820e981890.gif) | ![in-prompt](https://user-images.githubusercontent.com/5556114/113440545-46013480-940a-11eb-9e4b-5fb3b80c0d22.gif)

Thank you for the script kit. I'm already thinking about my next task to automate.

**Comment by johnlindquist** [comment]
Wow! Awesome scenario to automate away! ❤️❤️❤️

Love that you're finding it so helpful. Thank you so much for sharing and making my day!

**Reply by rvisharma** [reply]
Script kit indeed very cool. Loving it.
I just found out about `setPanel` api. That will be better suited instead of `setChoices` which i was using earlier to live update the status.
Also this opens up the possibility of previewing screenshots from playwright to the panel 🤯🚀

**Reply by johnlindquist** [reply]
Neat idea!

In the future, the plan is to use prompt only when asking for `arg` input. Other non-interactive/non-prompt visuals will use the [Script Widget](https://github.com/johnlindquist/kit/discussions/82) concept. Tiny windows that you have complete control over.

The prompt is more of a "blocking" UI modal that demands focus, whereas the Script Widget will be non-blocking.

**Reply by rvisharma** [reply]
sounds great!
